terraform/snowflake/modules/elt/roles.tf

```diff
@@ -224,22 +224,17 @@ resource "snowflake_grant_account_role" "streamlit_analytics_to_reporter" {
   parent_role_name = snowflake_account_role.reporter.name
 }
 
-locals {
-  streamlit_roles = {
-    analytics = snowflake_account_role.streamlit_analytics.name
-  }
-  databases = {
-    analytics = module.analytics.name
-  }
-}
-
-resource "snowflake_grant_privileges_to_account_role" "streamlit_privileges" {
-  provider          = snowflake.accountadmin
-  for_each          = local.streamlit_roles
-  account_role_name = each.value
-  privileges        = ["CREATE STREAMLIT", "CREATE STAGE"]
+resource "snowflake_grant_privileges_to_account_role" "streamlit_database_privileges" {
+  account_role_name = "${module.analytics.name}_STREAMLIT"
+  privileges        = ["CREATE STAGE"]
   on_account_object {
+    object_name = "ANALYTICS_PRD"
```

Comment:
This should use `module.analytics.name`, otherwise the dev environment won't work.

Reply:
The warning says "Interpolation-only expressions are deprecated in Terraform v0.12.14". I believe it could be fixed by just using `module.analytics.name` instead of having it in an interpolation string